cfg/config_test.go

```diff
@@ -52,6 +53,46 @@ func Test_Configuration_DefaultConfig(t *testing.T) {
 	assert.Contains(t, err.Error(), "operator namespace")
 }
 
+func Test_Configuration_GetGlobalFailedJobsHistoryLimit(t *testing.T) {
+	t.Run("GlobalKeepJobsIfNotSet", func(t *testing.T) {
+		c := NewDefaultConfig()
+		c.GlobalFailedJobsHistoryLimit = -1
+		c.GlobalKeepJobs = rand.Int()
```

Comment:
Be cautious with random values in tests. They tend to result in flappy tests. One failed a few passed and some fails and now good again. Use explicit test values that covers known edge cases (for example lower one out of bound, lower in bound, somewhere whithin the middle, upper in bound and upper out of bound).

There is fuzzy testing but that is a subject on its own and I would keep them in a testsuite on there own that is run separatley.

However, it should not be an issue here.